src/main/java/io/github/jsonSnapshot/SnapshotFile.java

```diff
@@ -54,23 +54,29 @@ private File createFile(String fileName) throws IOException {
     return file;
   }
 
-  public void push(String snapshot) {
-
-    rawSnapshots.add(snapshot);
-
-    File file = null;
+  public void push(@NonNull final SnapshotDataItem snapshot) {
+    storedSnapshots.add(snapshot);
 
+    final File file;
     try {
       file = createFile(fileName);
     } catch (IOException e) {
-      e.printStackTrace();
+      throw new RuntimeException("Unable to create snapshot items file " + fileName, e);
     }
 
-    try (FileOutputStream fileStream = new FileOutputStream(file, false)) {
-      byte[] myBytes = StringUtils.join(rawSnapshots, SPLIT_STRING).getBytes();
-      fileStream.write(myBytes);
+    final byte[] myBytes =
+        storedSnapshots
+            .getItems()
+            .stream()
+            .map(SnapshotDataItem::asRawData)
+            .collect(Collectors.joining(SPLIT_STRING))
+            .getBytes();
+
+    try {
+      Files.write(file.toPath(), myBytes);
```

Comment:
Maybe something like this? This would also close the file after writing.

```java
    final File file = createFile(fileName);  // exception handling inside
    try (FileWriter fileWriter = new FileWriter(file, false)) {
      String content = storedSnapshots.getItems().stream()
            .map(SnapshotDataItem::asRawData)
            .collect(Collectors.joining(SPLIT_STRING));
      fileWriter.write(content);
    } catch (IOException e) {
      // rethrow
    }
```

Reply:
done in https://github.com/json-snapshot/json-snapshot.github.io/pull/15/commits/863d2adecadc7d005918e0d152a1a8458d2117fc ,
and note that i also made sure we control the charset the file is read from written in.